code/modules/mob/living/simple_animal/hostile/megafauna/hierophant.dm

```diff
@@ -440,10 +393,114 @@ Difficulty: Hard
-	if(!istype(A) || get_dist(A, src) <= 2)
+/mob/living/simple_animal/hostile/megafauna/hierophant/Life()
+	. = ..()
+	if(. && spawned_beacon && !QDELETED(spawned_beacon) && !client)
+		if(target || loc == spawned_beacon.loc)
+			timeout_time = initial(timeout_time)
+		else
+			timeout_time--
+		if(timeout_time <= 0 && !did_reset)
+			did_reset = TRUE
+			visible_message("<span class='hierophant_warning'>\"Vixyvrmrk xs fewi...\"</span>")
+			blink(spawned_beacon)
+			adjustHealth(min((health - maxHealth) * 0.5, -250)) //heal for 50% of our missing health, minimum 10% of maximum health
+			wander = FALSE
+			if(health > maxHealth * 0.9)
+				visible_message("<span class='hierophant'>\"Vitemvw gsqtpixi. Stivexmrk ex qebmqyq ijjmgmirgc.\"</span>")
+			else
+				visible_message("<span class='hierophant'>\"Vitemvw gsqtpixi. Stivexmsrep ijjmgmirgc gsqtvsqmwih.\"</span>")
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/death()
+	if(health > 0 || hierophant_dying)
 		return
-	blink(A)
+	else
+		hierophant_dying = TRUE
+		blinking = TRUE
+		blinking = TRUE //we do a fancy animation, release a huge burst(), and leave our staff.
+		visible_message("<span class='hierophant'>\"Mrmxmexmrk wipj-hiwxvygx wiuyirgi...\"</span>")
+		visible_message("<span class='hierophant_warning'>[src] shrinks, releasing a massive burst of energy!</span>")
+		for(var/mob/living/L in view(7,src))
+			stored_nearby += L // store the people to grant the achievements to once we die
+		hierophant_burst(null, get_turf(src), 10)
+		..(force_grant = stored_nearby)
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/Destroy()
+	qdel(spawned_beacon)
+	. = ..()
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/devour(mob/living/L)
+	for(var/obj/item/W in L)
+		if(!L.dropItemToGround(W))
+			qdel(W)
+	visible_message("<span class='hierophant_warning'>\"[pick(kill_phrases)]\"</span>")
+	visible_message("<span class='hierophant_warning'>[src] annihilates [L]!</span>","<span class='userdanger'>You annihilate [L], restoring your health!</span>")
+	adjustHealth(-L.maxHealth*0.5)
+	L.dust()
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/CanAttack(atom/the_target)
+	. = ..()
+	if(istype(the_target, /mob/living/simple_animal/hostile/asteroid/hivelordbrood)) //ignore temporary targets in favor of more permanent targets
+		return FALSE
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/GiveTarget(new_target)
+	var/targets_the_same = (new_target == target)
+	. = ..()
+	if(. && target && !targets_the_same)
+		visible_message("<span class='hierophant_warning'>\"[pick(target_phrases)]\"</span>")
+		if(spawned_beacon && loc == spawned_beacon.loc && did_reset)
+			arena_trap(src)
+
+/mob/living/simple_animal/hostile/megafauna/hierophant/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(src && . > 0 && !blinking)
```

Comment:
does adjusthealth return the value healed/damaged? If it's just true...

```suggestion
	if(src && . && !blinking)
```

Reply:
also is the src check suppose to be `!QDELETED(src)`?

Reply:
hierophant code is a mess to be honest and i don't really want to touch it all that much because i'm afraid i'll totally fuck it up but i'll try that

Reply:
lol me in every file, especially in code i've written previously